### PR TITLE
Move whitelist to config file

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This enables other people deploying this to be able to use it :)

### CODE REVIEW

File: plugins/login/login.js [hunk 1]
```diff
@@ -44,7 +29,7 @@ module.exports = (config) => ({
             }
             const username = request.auth.credentials.profile.username;
 
-            if (!whitelist[username]) {
+            if (!config.whitelist.includes(username)) {
```
Comment (catto) [inline plugins/login/login.js]: Should this allow all users if the whitelist is empty or not configured?

**Comment by catto** [comment]
👍

File: plugins/login/login.js (hunk 1 above)
Comment (stjohnjohnson) [inline plugins/login/login.js]: Good point, I don't see why not.

**Comment by stjohnjohnson** [comment]
Okay, @catto I updated it.  I also fixed all the tests to use the native promises of Hapi.

File: test/plugins/login.test.js [hunk 2]
```diff
@@ -148,102 +148,136 @@ describe('login plugin test', () => {
...
+                            password
+                        });
+                    });
                 });
             });
         });
 
-        it('POST exists', (done) => {
+        it('POST exists', () => (
             server.inject({
```
Comment (catto) [inline test/plugins/login.test.js]: should be returned this?

Comment (stjohnjohnson) [inline test/plugins/login.test.js]: If i put `return server.inject` eslint will complain that it's redundant.  Instead I turned it into a short-hand return, see the `(` above this line.

File: test/plugins/login.test.js [hunk 3]
```diff
@@ -148,102 +148,136 @@ describe('login plugin test', () => {
...
-            });
-        });
+            })
+        ));
     });
 
     describe('POST /logout', () => {
-        it('exists', (done) => {
+        it('exists', () => (
             server.inject({
```
Comment (catto) [inline test/plugins/login.test.js]: same here

File: test/plugins/login.test.js [hunk 4]
```diff
@@ -148,102 +148,136 @@ describe('login plugin test', () => {
...
                 assert.notEqual(reply.statusCode, 404, 'Logout route should be available');
-                done();
-            });
-        });
+            })
+        ));
 
-        it('returns 200', (done) => {
+        it('returns 200', () => (
             server.inject({
```
Comment (catto) [inline test/plugins/login.test.js]: and same here

File: test/plugins/login.test.js (hunk 2 above)
Comment (catto) [inline test/plugins/login.test.js]: Oh I see! 🍡

**Comment by catto** [comment]
Thanks @stjohnjohnson LGTM 👍

**Comment by Filbird** [comment]
⬜ 📄 👍

**Pull request merged**